paddle/framework/op_registry.h

```diff
@@ -86,8 +106,70 @@ class OpProtoAndCheckerMaker {
     *(proto_->mutable_comment()) = comment;
   }
 
+ private:
+  void SetHasMultiple(const std::string& in_out, bool* flag) {
+    if (!*flag) {
+      AddAttr<std::vector<int>>(in_out + "_format",
+                                "The multiple index of " + in_out +
+                                    "\n"
+                                    R"DOC(
+This attribute is used by Paddle core framework. Paddle's Op support each input
+or output could be a list of variable. This attribute is used to show how that
+list organized.
+
+e.g.
+  input = ["a", "b", "c", "d", "e", "f"]
+  input_format = [0, 4, 5, 6]
+
+means
+  The all input variables of this op is six, and they are segmented into three
+  inputs.
+
+  The first input is input[0:4], second is input[4:5], third is input[5:6].
```

Comment:
That information is stored in `OpProto` because it is shared between all instances of the same types.

This PR does not touch C++ `OperatorBase` interface, but it is better for `OperatorBase` can get Input/Output information for that type.